File: lab04_stack/lab04_example02/src/lab04_example02_node.cpp
#include "ros/ros.h"
#include "tf2_ros/transform_listener.h"

int main(int argc, char **argv)
{
    ros::init(argc, argv, "robotis_tf_listener_node");

    ros::NodeHandle nh;

    ros::Rate rate(10);

    tf2_ros::Buffer tfBuffer;
    tf2_ros::TransformListener tfListener(tfBuffer);

    while (nh.ok())
    {
        geometry_msgs::TransformStamped T;

        try
        {
            T = tfBuffer.lookupTransform("world", "link6", ros::Time(0));
        }
        catch (tf2::TransformException &ex)
        {
            ROS_WARN("%s", ex.what());
            ros::Duration(1.0).sleep();
            continue;
        }

        std::cout << T << std::endl;

        rate.sleep();
    }
    return 52;
}

File: lab04/lab04_example02/scripts/lab04_example02_node.py
#!/usr/bin/env python

import rospy
import tf2_ros

if __name__ == '__main__':
    rospy.init_node('robotis_tf_listener_node')

    tfBuffer = tf2_ros.Buffer()
    listener = tf2_ros.TransformListener(tfBuffer)

    rate = rospy.Rate(10.0)

    while not rospy.is_shutdown():

        try:
            T = tfBuffer.lookup_transform('world', 'link6', rospy.Time(0))
        except (tf2_ros.LookupException, tf2_ros.ConnectivityException):
            rate.sleep()
            continue

        print T

        rate.sleep()